dev-docs/source/Architecture.rst

```diff
@@ -0,0 +1,180 @@
+============
+Architecture
+============
+
+.. contents::
+   :local:
+
+Summary
+-------
+
+Mantid is used for reduction and analysis of neutron and muon data.
+This section describes the architecture of the mantid software in terms
+of the `C4 model <C4Model_>`_.
+
+Context
+-------
+
+The mantid software interacts with people and other systems as described
+by the following context diagram.
+
+.. figure:: /images/architecture-context.png
+   :scale: 70%
+   :align: center
+   :alt: Context diagram of mantid software system
+
+   Context diagram for mantid software
+
+Experimental users of at neutron/muon facilities use the software
+to reduce/analyze their captured data. Mantid uses existing
+experimental data access systems to retrieve users' data
+for processing either
+
+- interactively via Python scripts or a GUI
+- automatically via programmatic access from other systems such as
+  automated reduction systems.
+
+Container
+---------
+
+*Please note that container here does not mean container in the
+deployment (Docker, Singularity) sense of the word, rather the generic
+definition of enclosing something.*
+
+The mantid software system is defined as four containers:
+two libraries (``framework`` & ``mantidqt``),
+one general-purpose graphical application (``workbench``) and
+many customized graphical interfaces for specific scientific techniques.
+This is illustrated, along with connections between the containers,
+in the following diagram.
+
+
+.. figure:: /images/architecture-container.png
+   :scale: 70%
+   :align: center
+   :alt: Container diagram of mantid software system
+
+   Container diagram for mantid software
```

Comment:
It's unclear if the workbench depends on the custom interfaces. They are not really necessary for workbench, still they are built when building the workbench and appear in the menus.

Reply:
They are not strict dependencies from the code side, they are basically plugins as we do the registration and discover at run time. I will try to make this clearer in the diagram.